src/test/java/uk/gov/hmcts/reform/wataskmonitor/services/CamundaServiceTest.java

```diff
@@ -0,0 +1,76 @@
+package uk.gov.hmcts.reform.wataskmonitor.services;
+
+import org.jetbrains.annotations.NotNull;
+import org.json.JSONException;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.skyscreamer.jsonassert.JSONAssert;
+import org.skyscreamer.jsonassert.JSONCompareMode;
+import uk.gov.hmcts.reform.wataskmonitor.clients.CamundaClient;
+import uk.gov.hmcts.reform.wataskmonitor.models.CamundaTask;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class CamundaServiceTest {
+
+    public static final String SERVICE_TOKEN = "some service token";
+    @Mock
+    private CamundaClient camundaClient;
+
+    @InjectMocks
+    private CamundaService camundaService;
+
+    @Captor
+    private ArgumentCaptor<String> actualQueryParametersCaptor;
+
+    @Test
+    void givenGetTasksCamundaRequestShouldRetrieveTasks() throws JSONException {
+        List<CamundaTask> expectedCamundaTasks = List.of(new CamundaTask("1"), new CamundaTask("2"));
+        when(camundaClient.getTasks(
+            eq(SERVICE_TOKEN),
+            eq("0"),
+            eq("1000"),
+            actualQueryParametersCaptor.capture()
+        )).thenReturn(expectedCamundaTasks);
+
+        List<CamundaTask> actualCamundaTasks = camundaService.getUnConfiguredTasks(SERVICE_TOKEN);
+
+        JSONAssert.assertEquals(
+            getExpectedQueryParameters(),
+            actualQueryParametersCaptor.getValue(),
+            JSONCompareMode.LENIENT
+        );
+
+        assertThat(actualCamundaTasks).isEqualTo(expectedCamundaTasks);
+    }
+
```

Comment:
should we introduce a test to prove delayed tasks are not retrieved s part fo this query?

Reply:
done